Add a pre-render subscriber to custom block (#2332)
ibexa/documentation-developer: docs/content_management/pages/create_custom_page_block.md

    
    If you consider using React JavaScript library, see [React App block](react_app_block.md).

## Add pre-render event listener

> If you need to compute variable to pass to the template, you can listen or subscribe to the block pre-render event.

Reviewer: ```suggestion
If you need to compute a variable to pass to the template, you can listen or subscribe to the block pre-render event.
```

Author: @dabrt ?
```suggestion
If you need to compute variables to pass to the template, you can listen or subscribe to the block pre-render event.
```

> For example, the following event subscriber load the event content item and pass it to the template as `event_content`:

Reviewer: ```suggestion
For example, the following event subscriber loads the `event` content item and passes it to the template as `event_content`:
```

``` php
[[= include_file('code_samples/page/custom_page_block/src/Event/Subscriber/BlockEmbedEventEventSubscriber.php') =]]
```

> Now, block view template could use `ibexa_render(event_content, {'viewType': 'embed'})` instead of `render(controller('ibexa_content::viewAction', {'contentId': event, 'viewType': 'embed'}))`,
> other [content Twig functions](content_twig_functions.md), or [field Twig functions](field_twig_functions.md).

Reviewer: ```suggestion
The block view template could now use `ibexa_render(event_content, {'viewType': 'embed'})` instead of `render(controller('ibexa_content::viewAction', {'contentId': event, 'viewType': 'embed'}))`, other [content Twig functions](content_twig_functions.md), or [field Twig functions](field_twig_functions.md).
```
